components/camera/client_test.go

```diff
@@ -483,16 +496,36 @@ func TestClientLazyImage(t *testing.T) {
 	camera1Client, err := camera.NewClientFromConn(context.Background(), conn, "", camera.Named(testCameraName), logger)
 	test.That(t, err, test.ShouldBeNil)
 
-	ctx := gostream.WithMIMETypeHint(context.Background(), rutils.MimeTypePNG)
-	frame, _, err := camera.ReadImage(ctx, camera1Client)
+	// TODO(hexbabe): remove below test when Stream/ReadImage pattern is refactored
+	t.Run("lazily decode from ReadImage without lazy suffix", func(t *testing.T) {
+		ctx := gostream.WithMIMETypeHint(context.Background(), rutils.MimeTypePNG)
+		frame, _, err := camera.ReadImage(ctx, camera1Client)
+		test.That(t, err, test.ShouldBeNil)
+		// Should always lazily decode
+		test.That(t, frame, test.ShouldHaveSameTypeAs, &rimage.LazyEncodedImage{})
+		frameLazy := frame.(*rimage.LazyEncodedImage)
+		test.That(t, frameLazy.RawData(), test.ShouldResemble, imgBuf.Bytes())
+	})
+
+	ctx := context.Background()
+	frame, err := camera.DecodeImageFromCamera(ctx, rutils.MimeTypePNG, nil, camera1Client)
 	test.That(t, err, test.ShouldBeNil)
 	// Should always lazily decode
 	test.That(t, frame, test.ShouldHaveSameTypeAs, &rimage.LazyEncodedImage{})
 	frameLazy := frame.(*rimage.LazyEncodedImage)
 	test.That(t, frameLazy.RawData(), test.ShouldResemble, imgBuf.Bytes())
 
-	ctx = gostream.WithMIMETypeHint(context.Background(), rutils.WithLazyMIMEType(rutils.MimeTypePNG))
-	frame, _, err = camera.ReadImage(ctx, camera1Client)
+	// TODO(hexbabe): remove below test when Stream/ReadImage pattern is refactored
+	t.Run("lazily decode from ReadImage", func(t *testing.T) {
+		ctx = gostream.WithMIMETypeHint(context.Background(), rutils.WithLazyMIMEType(rutils.MimeTypePNG))
+		frame, _, err = camera.ReadImage(ctx, camera1Client)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, frame, test.ShouldHaveSameTypeAs, &rimage.LazyEncodedImage{})
+		frameLazy = frame.(*rimage.LazyEncodedImage)
```

Comment:
shouldn't we check https://github.com/viamrobotics/rdk/pull/4487/files#diff-7ee787cbf724e27845fd712b8432895502746bc4f6fa0ea7a03c20a1a754e8daR507?

Reply:
I don't know if the link is opening properly. It just takes me back to the top of the PR diff

Reply:
shouldn't we check the content of the image?

Reply:
Ah I understand now ya. This is a terrible test as a result of me trying to add back a previously deleted `ReadImage` test. Let me fix i.e. include checking content